cmd/gardenlet/app/gardenlet.go

```diff
@@ -608,3 +615,38 @@ func extractID(line string) string {
 
 	return id
 }
+
+func checkSeedConfig(ctx context.Context, seedClient client.Client, seedConfig *config.SeedConfig) error {
+	if seedConfig == nil {
+		// Nothing to check
+		return nil
+	}
+
+	shootInfo := &corev1.ConfigMap{}
+	if err := seedClient.Get(ctx, kutil.Key(metav1.NamespaceSystem, v1beta1constants.ConfigMapNameShootInfo), shootInfo); client.IgnoreNotFound(err) != nil {
+		return err
+	} else if apierrors.IsNotFound(err) {
+		// Seed cluster does not seem to be managed by Gardener
+		return nil
+	}
+
+	podNetwork := shootInfo.Data["podNetwork"]
+	if podNetwork != seedConfig.Spec.Networks.Pods {
+		return fmt.Errorf("incorrect pod network specified in seed configuration (cluster=%q vs. config=%q)", podNetwork, seedConfig.Spec.Networks.Pods)
+	}
+
+	serviceNetwork := shootInfo.Data["serviceNetwork"]
+	if serviceNetwork != seedConfig.Spec.Networks.Services {
+		return fmt.Errorf("incorrect service network specified in seed configuration (cluster=%q vs. config=%q)", serviceNetwork, seedConfig.Spec.Networks.Services)
+	}
+
+	nodeNetwork, exists := shootInfo.Data["nodeNetwork"]
+	// Be graceful in case the (optional) node network is only available on one side
+	if seedConfig.Spec.Networks.Nodes != nil && exists {
+		if nodeNetwork != *seedConfig.Spec.Networks.Nodes {
```

Comment:
Cosmetics:

```suggestion
	if nodeNetwork, exists := shootInfo.Data["nodeNetwork"]; exists &&
		seedConfig.Spec.Networks.Nodes != nil &&
		*seedConfig.Spec.Networks.Nodes != nodeNetwork {
		// Be graceful in case the (optional) node network is only available on one side
```

Reply:
Addressed in latest commit.